File: src/MaksIT.WebUI/src/models/identity/user/PatchUserEntityScopeRequest.ts
import { PatchRequestModelBase } from '../../PatchRequestModelBase'
import { ScopeEntityType, ScopePermission } from '../../engine/scopeEnums'

/** Mirrors <c>MaksIT.CertsUI.Models.Identity.User.PatchUserEntityScopeRequest</c>. */
export interface PatchUserEntityScopeRequest extends PatchRequestModelBase {
  id?: string | null
  entityId?: string | null
  entityType?: ScopeEntityType | null
  scope?: ScopePermission | null
}

File: src/MaksIT.CertsUI/Models/Identity/User/PatchUserEntityScopeRequest.cs
using MaksIT.CertsUI.Engine;
using MaksIT.Core.Abstractions.Webapi;


namespace MaksIT.CertsUI.Models.Identity.User;

public class PatchUserEntityScopeRequest : PatchRequestModelBase {
  public Guid? Id { get; set; }

  /// <summary>
  /// The ID of the entity (e.g., organization or application). Optional for update-in-place when only Id is sent.
  /// </summary>
  public Guid? EntityId { get; set; }

  /// <summary>
  /// The type of the entity (e.g., "Organization", "Application"). Optional for update-in-place when only Id is sent.
  /// </summary>
  public ScopeEntityType? EntityType { get; set; }

  /// <summary>
  /// The scope granted to the user for this entity. Optional for update-in-place when only Id is sent.
  /// </summary>
  public ScopePermission? Scope { get; set; }
}